content/docs/tasks/policy-enforcement/denial-and-list/index.md

```diff
@@ -160,23 +163,103 @@ Save the following YAML snippet as `checkversion-rule.yaml`:
 1. Verify that when you access the Bookinfo `productpage` (`http://$GATEWAY_URL/productpage`) without logging in, you see **no** stars.
 Verify that after logging in as "jason" you see black stars.
 
+## IP-based _whitelists_ or _blacklists_
+
+Istio supports _whitelists_ and _blacklists_ based on IP address. You can
+configure Istio to accept or reject requests from a specific IP address or a
+subnet.
+
+1. Verify you can access the Bookinfo `productpage` found at
+   `http://$GATEWAY_URL/productpage`. You won't be able to access it once you
+   apply the rules below.
+
+1.  Create configuration for the [list](/docs/reference/config/policy-and-telemetry/adapters/list/)
+    adapter that lists subnet `"10.57.0.0\16"`.
+    Save the following YAML snippet as `whitelistip-handler.yaml`:
+
+    {{< text yaml >}}
+    apiVersion: config.istio.io/v1alpha2
+    kind: listchecker
+    metadata:
+      name: whitelistip
+    spec:
+      # providerUrl: ordinarily black and white lists are maintained
+      # externally and fetched asynchronously using the providerUrl.
+      overrides: ["10.57.0.0/16"]  # overrides provide a static list
+      blacklist: false
+      entryType: IP_ADDRESSES
+    {{< /text >}}
+
+1. After saving the snippet, run the following command:
+
+    {{< text bash >}}
+    $ kubectl apply -f whitelistip-handler.yaml
+    {{< /text >}}
+
+1.  To extract the source IP, create a [list entry instance](/docs/reference/config/policy-and-telemetry/templates/listentry/)
+    from the template. You can use the request headers `x-forwarded-for` or
+    `x-real-ip` as needed. Save the following YAML snippet as
+    `sourceip-instance.yaml`:
+
+    {{< text yaml >}}
+    apiVersion: config.istio.io/v1alpha2
+    kind: listentry
+    metadata:
+      name: sourceip
+    spec:
+      value: source.ip | ip("0.0.0.0")
+    {{< /text >}}
+
+1. After saving the snippet, run the following command:
+
+    {{< text bash >}}
+    $ kubectl apply -f sourceip-instance.yaml
+    {{< /text >}}
+
+1.  Enable `whitelist` checking for the ratings service. Save the following
+    YAML snippet as `checkip-rule.yaml`:
+
+    {{< text yaml >}}
+    apiVersion: config.istio.io/v1alpha2
+    kind: rule
+    metadata:
+      name: checkip
+    spec:
+      match: source.labels["istio"] == "ingressgateway"
+      actions:
+      - handler: whitelistip.listchecker
+        instances:
+        - sourceip.listentry
+    {{< /text >}}
+
+1. After saving the snippet, run the following command:
+
+    {{< text bash >}}
+    $ kubectl apply -f checkip-rule.yaml
+    {{< /text >}}
+
+1. Try to access the Bookinfo `productpage` at
+   `http://$GATEWAY_URL/productpage` and verify that you get an error similar
+   to: `PERMISSION_DENIED:staticversion.istio-system:<your mesh source ip> is
+   not whitelisted`
+
 ## Cleanup
 
-*   Remove the mixer configuration:
+* Remove the mixer configuration:
 
     {{< text bash >}}
-    $ kubectl delete -f checkversion-rule.yaml
```

Comment:
This part of cleanup is different for attribute based and ip based tasks as the yamls they are deleting are different.

We can have one common cleanup which cleans up routing rules, destination rules etc and a cleanup specific to tasks where the cleanup of the yamls they are applying is happening?